src/efi.rs

```diff
@@ -365,23 +329,94 @@ impl Component for Efi {
             self.mount_esp_device(Path::new(dest_root), Path::new(&esp_device))?
         };
 
+        let destpath_clone = destpath.clone();
+
         let destd = &openat::Dir::open(&destpath)
             .with_context(|| format!("opening dest dir {}", destpath.display()))?;
         validate_esp_fstype(destd)?;
 
+        // let using_systemd_boot = skip_systemd_bootloaders();
+        let using_systemd_boot = true;
+
         // TODO - add some sort of API that allows directly setting the working
         // directory to a file descriptor.
         std::process::Command::new("cp")
             .args(["-rp", "--reflink=auto"])
             .arg(&srcdir_name)
-            .arg(destpath)
+            .arg(&destpath)
             .current_dir(format!("/proc/self/fd/{}", src_root.as_raw_fd()))
             .run()?;
-        if update_firmware {
+
+        // Configure systemd-boot loader config/entry
+        if using_systemd_boot {
+            let loader_dir = destpath.join("loader");
+            if !loader_dir.exists() {
+                log::warn!("Creating systemd-boot loader directory at {}", loader_dir.display());
+                std::fs::create_dir_all(&loader_dir)
+                    .with_context(|| format!("creating systemd-boot loader directory {}", loader_dir.display()))?;
+            }
+            let entries_dir = loader_dir.join("entries");
+            log::warn!("Using systemd-boot, creating entries dir at {}", entries_dir.display());
+            std::fs::create_dir_all(&entries_dir)
+                .with_context(|| format!("creating entries dir {}", entries_dir.display()))?;
+            log::warn!("Installing systemd-boot entry in {}/bootupd.conf", entries_dir.display());
+            // Write loader.conf if it doesn't exist
+            let loader_conf_path = loader_dir.join("loader.conf");
+            if !loader_conf_path.exists() {
+                let mut loader_conf = std::fs::File::create(&loader_conf_path)
+                    .with_context(|| format!("creating loader.conf at {}", loader_conf_path.display()))?;
+                writeln!(loader_conf, "default auto")?;
```

Comment:
This one should probably come from the container image and not be hardcoded in bootupd.